packages/gestalt/src/InternalDismissButton.js

```diff
@@ -0,0 +1,108 @@
+// @flow strict
+import { type Node, forwardRef, useImperativeHandle, useState, useRef } from 'react';
+import classnames from 'classnames';
+import Pog from './Pog.js';
+import { type AbstractEventHandler } from './AbstractEventHandler.js';
+import styles from './IconButton.css';
+import touchableStyles from './Touchable.css';
+import useFocusVisible from './useFocusVisible.js';
+import useTapFeedback from './useTapFeedback.js';
+
+type Props = {|
+  accessibilityLabel: string,
+  accessibilityControls?: string,
+  onClick?: AbstractEventHandler<
+    SyntheticMouseEvent<HTMLButtonElement> | SyntheticKeyboardEvent<HTMLButtonElement>,
+  >,
+|};
+
+const InternalDismissIconButtonWithForwardRef: React$AbstractComponent<Props, HTMLButtonElement> =
+  forwardRef<Props, HTMLButtonElement>(function IconButton(props: Props, ref): Node {
+    const { accessibilityLabel, accessibilityControls, onClick } = props;
+
+    const innerRef = useRef(null);
+    // When using both forwardRef and innerRef, React.useimperativehandle() allows a parent component
+    // that renders <IconButton ref={inputRef} /> to call inputRef.current.focus()
+    useImperativeHandle(ref, () => innerRef.current);
+
+    const {
+      compressStyle,
+      isTapping,
+      handleBlur,
+      handleMouseDown,
+      handleMouseUp,
+      handleTouchStart,
+      handleTouchMove,
+      handleTouchCancel,
+      handleTouchEnd,
+    } = useTapFeedback({
+      height: innerRef?.current?.clientHeight,
+      width: innerRef?.current?.clientWidth,
+    });
+
+    const [isFocused, setFocused] = useState(false);
+    const [isHovered, setHovered] = useState(false);
+
+    const { isFocusVisible } = useFocusVisible();
+
+    const renderPogComponent = (selected?: boolean): Node => (
+      <Pog
+        focused={isFocusVisible && isFocused}
+        hovered={isHovered}
+        icon="cancel"
+        selected={selected}
+        size="lg"
```

Comment:
This component is always the same size? Do we need to support other sizes?

Reply:
I'd keep it lg for now, We can support other sizes if neede.d